src/tabletraits.jl

```diff
@@ -46,3 +46,25 @@ function NDSparse(x; idxcols::Union{Void,Vector{Symbol}}=nothing, datacols::Unio
         throw(ArgumentError("x cannot be turned into an NDSparse."))
     end
 end
+
+function table(rows::AbstractArray{T}; kwargs...) where {T<:NamedTuple}
+    source_data, source_names = TableTraitsUtils.create_columns_from_iterabletable(rows)
+    
+    kwargs_dict = Dict(i[1]=>i[2] for i in kwargs)   
+    kwargs_dict[:copy] = false
+
+    return table(source_data..., names=source_names; kwargs_dict...)
+end
```

Comment:
This is ambiguous with the constructor for a table with 1 column which are named tuples. Normally I'm not a big fan of special cases like this, but this one might just be OK.